pkg/inventory/inventory-client.go

```diff
@@ -377,9 +394,15 @@ func (cic *ClusterInventoryClient) createInventoryObj(obj *unstructured.Unstruct
 	}
 
 	klog.V(4).Infof("creating inventory object: %s/%s", obj.GetNamespace(), obj.GetName())
-	_, err = cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
-		Create(context.TODO(), obj, metav1.CreateOptions{})
-	return err
+	var createdObj *unstructured.Unstructured
+	if createdObj, err = cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
+		Create(context.TODO(), obj, metav1.CreateOptions{}); err != nil {
+		return err
+	}
+	if createdObj != nil {
+		cic.invResourceVersion = createdObj.GetResourceVersion()
```

Comment:
Why not just return the createdObj and copy the ResourceVersion in the caller? Caching it on the client struct seems unnecessary.

Reply:
We can also do it that way. Caching it looks simpler implementation. If we return the createdObj, we need to update multiple places in the code.

Reply:
It's best not to cache things on the struct if you don't need to. It makes the code thread unsafe. The private methods are not exposed, so changing their signature is fine.

Reply:
done